pkg/verifier/notation/notation.go

```diff
@@ -214,25 +230,48 @@ func parseVerifierConfig(verifierConfig config.VerifierConfig, namespace string)
 	return conf, nil
 }
 
+// convert <store>:<certs> to ca:<store><certs>
+func verificationCertStoresConversion(conf *NotationPluginVerifierConfig) error {
+	storeTypeAsKeyCount := 0
+	certStoresLen := len(conf.VerificationCertStores)
+	for _, storeType := range trustStoreTypes {
+		if _, ok := conf.VerificationCertStores[storeType]; ok {
+			storeTypeAsKeyCount++
+		}
+	}
+
+	if storeTypeAsKeyCount == 0 {
+		conf.VerificationCertStores = map[string]any{
+			typeCA: conf.VerificationCertStores,
+		}
+	} else if certStoresLen > storeTypeAsKeyCount {
+		return re.ErrorCodeConfigInvalid.NewError(re.Verifier, conf.Name, re.EmptyLink, nil, fmt.Sprintf("failed to parse to VerificationCertStores from: %+v. which using a mix of certStoresByType and legacy certStores", conf.VerificationCertStores), re.HideStackTrace)
+	}
+	return nil
+}
+
 // signatures should not have nested references
 func (v *notationPluginVerifier) GetNestedReferences() []string {
 	return []string{}
 }
 
 // append namespace to certStore so they are uniquely identifiable
-func prependNamespaceToCertStore(verificationCertStore map[string][]string, namespace string) (map[string][]string, error) {
+func prependNamespaceToCertStore(verificationCertStores map[string]interface{}, namespace string) (map[string]interface{}, error) {
 	if namespace == "" {
 		return nil, re.ErrorCodeEnvNotSet.WithComponentType(re.Verifier).WithDetail("failure to parse VerificationCertStores, namespace for VerificationCertStores must be provided")
 	}
-
-	for _, certStores := range verificationCertStore {
-		for i, certstore := range certStores {
-			if !isNamespacedNamed(certstore) {
-				certStores[i] = namespace + constants.NamespaceSeperator + certstore
+	for _, trustStoreType := range trustStoreTypes {
+		if mapValue, ok := verificationCertStores[trustStoreType].(map[string]interface{}); ok {
+			for _, certStores := range mapValue {
+				for i, certstore := range certStores.([]interface{}) {
+					if !isNamespacedNamed(certstore.(string)) {
```

Comment:
What if `certStores` is not `[]interface{}`? or can it not be `[]interface{}`? Will it panic?
Same question for `certStore.(string)`.

Reply:
Have you tested it E2E?

Reply:
concrete types have been placed instead of interface